google/cloud/internal/url_encode.cc

```diff
@@ -24,15 +24,31 @@ namespace internal {
 std::string UrlEncode(absl::string_view value) {
   using CharMapping = std::pair<absl::string_view, absl::string_view>;
   auto const mappings = std::array<CharMapping, 25>{{
-      {" ", "%20"}, {"\"", "%22"}, {"#", "%23"},  {"$", "%24"}, {"%", "%25"},
-      {"&", "%26"}, {"+", "%2B"},  {",", "%2C"},  {"/", "%2F"}, {":", "%3A"},
-      {";", "%3B"}, {"<", "%3C"},  {"=", "%3D"},  {">", "%3E"}, {"?", "%3F"},
-      {"@", "%40"}, {"[", "%5B"},  {"\\", "%5C"}, {"]", "%5D"}, {"^", "%5E"},
-      {"`", "%60"}, {"{", "%7B"},  {"|", "%7C"},  {"}", "%7D"}, {"\177", "%7F"},
+      {" ", "%20"}, {"\"", "%22"}, {"#", "%23"}, {"$", "%24"},    {"&", "%26"},
```

Comment:
This just moves the % matcher to the end it is the same as UrlDeconde.

Reply:
See below.